src/endpoint/endpoint.js

```diff
@@ -210,6 +209,16 @@ async function main(options = {}) {
     }
 }
 
+async function is_http_allowed(nsfs_config_root) {
+    const allow_http = await endpoint_utils.get_nsfs_system_property('allow_http', nsfs_config_root);
+        if (nsfs_config_root && allow_http === undefined) {
```

Comment:
214 and 216 conditions can be combined,
something like `if (nsfs_config_root && allow_http === undefined && allow_http)`, since both conditions returing same value. Is n't it ?